### PR TITLE
Implement HUD and player_lives functionality

### PR DESCRIPTION
**Pull request by Derek52**
I have added a HUD scene and script to the Spaceship folder.
I just used one of the button graphics as a placeholder sprite, so someone will need to change that.

I've implemented a pretty customizable system that allows for gaining and losing lives.
There are methods called remove_life() and add_life(). The idea is that the spaceship, or level scene, will have signals that connect to the HUD and call those methods when the player takes damage. There are a few different ways to architect this.

My script has a MAX_Lives variable, which is the number of graphic to display. I set it to 4, like in the example you showed. It will actually give the player 5 lives though. One life that goes with each graphic, and one final life when those 4 are gone. It will be pretty easy to change that behavior if desired.

The scene has one Sprite node. Simply set the sprite resource on that node to the health_icon, and then put in a few position variables into my script, and the script will generate the rest of the graphics, and position them automatically.

I could have put 5 sprite nodes in the scene, but that'd be a lot of hardcoding and would be ugly.

If I made any design decisions you didn't like, or you need something amended, just let me know, and I'll try to fix it soon.

### CODE REVIEW

File: Scenes/Main.tscn [hunk 1]
```diff
@@ -94,11 +94,16 @@ scale = Vector2( 4, 4 )
 
 [node name="AnimationPlayer" type="AnimationPlayer" parent="Spaceship"]
 anims/spaceshipArrives = SubResource( 1 )
+<<<<<<< HEAD
```
Comment (davcri) [inline Scenes/Main.tscn]: It seems that the merge was not resolved correctly. Did you ever had this problem? Do you know how to fix it?

**Comment by davcri** [comment]
Hi @Derek52 thanks for your interest in the project! The system that you designed seems pretty solid, but I still have to look at the code.

I left a review on the PR: something went wrong and the scene files are corrupted (they contains git metadata for conflicting files). Let me know if you want to fix this, otherwise I can do it for you!

**Comment by davcri** [comment]
Sure!
What I would do is this:
Install Visual Studio Code if you don't have it (it's good to work with git).

1. Open the project with VSCode
2. Find all the unresolved conflicts by pressing CTRL+SHIFT+F and searching for `HEAD`
You will have a list of occurences on the left panel. Something like:
Screenshot_20191008_212551
3. **For each unresolved conflict** use the editor to accpet one change or another. After this step you should not see any text like  
```
<<<<< HEAD
=====
>>>>>
````
So for the previous example:
Screenshot_20191008_213003
4. Now save all files, create a commit and push it to update this PR :smile:

**Comment by Derek52** [comment]
I've pushed a new commit, that I believe solves the problems. I wasn't sure what changes to accept in the main scene. I think I made the right choice though.
Also Visual Studio code is nice, but I used grep to search for issues, and fixed them in Vim.

File: project.godot [hunk 2]
```diff
@@ -19,10 +19,6 @@ config/name="Space Shooter"
 run/main_scene="res://Scenes/Main.tscn"
 config/icon="res://icon.png"
 
-[autoload]
```
Comment (davcri) [inline project.godot]: We need this for the autoload of the scripts

**Comment by davcri** [comment]
Hi @Derek52 I just merged your HUD scene! You fixed the merge successfully :+1: Also good choice of tools with vim and grep! I would suggest using `rg` instead of `grep` and `nvim` instead of `vim`, check those tools (they are the same, but better)!

Regarding the PR: I just updated the coding style a little bit and I left a comment on the review section.

**Pull request merged**